### PR TITLE
Add extra volumes and volume mounts on gateway proxy for helm

### PR DESCRIPTION
**Pull request by kevin-shelaga**
# Description
Add helm values and tests for extra volume and volume mounts on the gateway proxy

Please include a summary of the changes.

Add helm values to address https://github.com/solo-io/gloo/issues/4198, including values, tests and a changelog

# Context

This will allow users to more easily mount sslFiles without the use of Kustomize

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/4198

### CODE REVIEW

**Comment by kevin-shelaga** [comment]
I am not a big fan of how the docs were generated for the new helm values, open to suggestions

File: install/helm/gloo/generate/values.go [hunk 1]
```diff
@@ -281,11 +281,13 @@ type GatewayProxy struct {
 	ExtraEnvoyArgs                 []string                     `json:"extraEnvoyArgs,omitempty" desc:"envoy container args, (e.g. https://www.envoyproxy.io/docs/envoy/latest/operations/cli)"`
 	ExtraContainersHelper          string                       `json:"extraContainersHelper,omitempty"`
 	ExtraInitContainersHelper      string                       `json:"extraInitContainersHelper,omitempty"`
+	ExtraVolumes                   []*appsv1.Volume             `json:"extraVolumes,omitempty"`
```
Comment (kdorosh) [inline install/helm/gloo/generate/values.go:284]: if we used a `map[string]interface{}` as a type here, there would be much fewer auto-generated docs for the helm value

File: install/test/helm_test.go [hunk 2]
```diff
@@ -1675,6 +1675,23 @@ var _ = Describe("Helm Test", func() {
...
+						})
+
+						prepareMakefile(namespace, helmValues{
+							valuesArgs: []string{
+								"gatewayProxies.gatewayProxy.ExtraProxyVolumeMounts[0].name=tls-crt",
+								"gatewayProxies.gatewayProxy.ExtraProxyVolumeMounts[0].mountPath=/certs/crt",
+								"gatewayProxies.gatewayProxy.ExtraProxyVolumeMounts[0].readOnly=true",
+							},
+						})
+
```
Comment (kdorosh) [inline install/test/helm_test.go:1694]: when do we assert that these changes are present in the rendered chart? I don't see any changes to the chart in this PR

**Review by kdorosh — CHANGES_REQUESTED** [review]
no changes to chart templates were made

File: install/helm/gloo/generate/values.go (hunk 1 above)
Comment (kevin-shelaga) [reply install/helm/gloo/generate/values.go:284]: agreed, will make this change

File: install/test/helm_test.go (hunk 2 above)
Comment (kevin-shelaga) [reply install/test/helm_test.go:1694]: hmm, for some reason my changes to 7-gateway-proxy-deployment.yaml didnt get pushed. Will push again with the values mapping for docs change

Comment (kdorosh) [reply install/test/helm_test.go:1694]: where in the tests do we assert that these values were picked up? i don't see any new assertions here on the PR

**Comment by kevin-shelaga** [comment]
@kdorosh Pr updated

File: install/test/helm_test.go (hunk 2 above)
Comment (kdorosh) [reply install/test/helm_test.go:1694]: bump (still don't see any new assertions in the test code)

**Comment by kevin-shelaga** [comment]
@kdorosh I was sort of following along with how "gatewayProxies.gatewayProxy.extraVolumeHelper" was implemented. It seems like testManifest.ExpectDeploymentAppsV1(gatewayProxyDeployment) is just testing the whole deployment spec. Separate Volume and VolumeMount tests are probably best. 

https://github.com/solo-io/gloo/blob/e856fd99befe67925bd7840b14ce2a1551c5ca60/install/test/helm_test.go#L1678

File: install/helm/gloo/templates/7-gateway-proxy-deployment.yaml [hunk 3]
```diff
@@ -179,9 +179,9 @@ spec:
           hostPort: {{ $spec.podTemplate.httpsPort }}
           {{- end}}
           {{- end}}
-{{- with $spec.podTemplate.extraPorts }}
-{{toYaml . | indent 8}}{{- end }}
-{{- if $spec.podTemplate.resources }}
+          {{- with $spec.podTemplate.extraPorts }}
+          {{ toYaml . | indent 8}}{{- end }}
+          {{- if $spec.podTemplate.resources }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/7-gateway-proxy-deployment.yaml:184]: why change this?

Comment (kevin-shelaga) [reply install/helm/gloo/templates/7-gateway-proxy-deployment.yaml:184]: @kdorosh I found the "{{toYaml" to be inconsistent with other "{{ toYaml" with the space, indentation must of been a local linting thing. Should i change it back?

**Review by kdorosh — DISMISSED** [review]

**Comment by kdorosh** [comment]
Unfortunately given the latest release, we now need to update the changelog here

**Comment by kevin-shelaga** [comment]
added v1.7.0-beta14 folder and moved changelog there

**Review by kdorosh — APPROVED** [review]
thanks for the contribution! 🚀

**Comment by kevin-shelaga** [comment]
Thank you for your help @kdorosh

**Pull request merged**